joblib/memory.py

```diff
@@ -402,13 +402,27 @@ class MemorizedFunc(Logger):
     verbose: int, optional
         The verbosity flag, controls messages that are issued as
         the function is evaluated.
+
+    hash_func: callable, optional
+        Parameters: (obj: object, hash_name: string, coerce_mmap: Boolean)
+        Return the hash from an object. Default is None.
+        If hash_func is None hashing.hash is used
     """
     # ------------------------------------------------------------------------
     # Public interface
     # ------------------------------------------------------------------------
 
-    def __init__(self, func, location, backend='local', ignore=None,
-                 mmap_mode=None, compress=False, verbose=1, timestamp=None):
+    def __init__(
+            self,
+            func,
+            location,
+            backend='local',
+            ignore=None,
+            mmap_mode=None,
+            compress=False,
+            verbose=1,
+            timestamp=None,
+            hash_func=None):
```

Comment:
To be more consistent with other function, I would keep it as it was previously.

```suggestion
    def __init__(self, func, location, backend='local', ignore=None,
                        mmap_mode=None, compress=False, verbose=1,
                        timestamp=None, hash_func=None):
```